### DISCUSSION TITLE
Supabase/other DB backend Support

### DISCUSSION THREAD

**Discussion by curiouscod3** [vriteio/vrite#53] [Ideas]
Hi, 
For now, It looks like only support git sync.
I am curious if vrite can support Supabase/ or other backends in the future to publish/sync mdx content through vrite editor.
I am wondering if It is possible to create an api server to sync a supabase or other database backend.
There is no documentary/tutorial in terms of API/Webhook in detail tho.

**Comment by areknawo** [comment]
I'm not sure if you want to self-host Vrite with Supabase as a backend or sync content from Vrite to your Supabase app. So, I'll cover both cases.

**Self-hosting Vrite with Supabase**
Vrite has a "custom" Node.js backend and uses MongoDB as its database. It's all pretty tightly integrated right now to allow real-time collaboration and power the API.

So, if you want to use Supabase as a backend - it's not possible right now and is not on the roadmap. That said, you should be able to use Docker Compose and self-host it on most cloud providers. You can check out the early self-hosting guide [here](https://docs.vrite.io/self-hosting/docker/).

**Syncing with Supabase**
If you'd like to sync content to your Supabase app (I don't quite understand your use case), you can use Vrite's API, Webhooks, and probably the JS SDK. The rest will depend on the specifics of your app.

**API, Webhooks, JS SDK**
Currently, Vrite's Webhooks aren't very advanced and are [documented here](https://docs.vrite.io/usage-guide/configuring-vrite/#managing-webhooks). Vrite API is documented with [Swagger](https://generator.swagger.io/?url=https://api.vrite.io/swagger.json), while the [JS SDK](https://docs.vrite.io/javascript-sdk/introduction/) is closely mapped with the API and has TypeScript typings, so it should be fairly easy to find your way around.

I know the docs need some more love, and I hope to focus more on them soon. I also plan to add new Webhooks, and more features to them (e.g. verification, grouping, throttling, etc.).
